### PR TITLE
Do `autocast` only on forward

### PR DESCRIPTION
**Pull request by emcastillo**
Since torch 1.8 the autocast & checkpointing seems to be fixed, so we apply it only to the forward pass.

### CODE REVIEW

File: pytorch_pfn_extras/handler.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def torch_autocast(enabled=True):
         yield
 
 
+_is_torch_17 = torch.__version__.split('.')[1] == '7'
```
Comment (emcastillo) [inline pytorch_pfn_extras/handler.py:30]: This is horrible, minimum supported is 1.7 so I just check against it like this ...

**Comment by kmaehashi** [comment]
As we now dropped support for PyTorch 1.7 (#307) let's remove the version check :smile:
Also please resolve conflicts :pray:

**Comment by emcastillo** [comment]
PTAL!

**Comment by kmaehashi** [comment]
Could you resolve conflicts and update docstrings? (`autocast` config moved to Logic)

**Comment by emcastillo** [comment]
rebased! PTAL!

**Comment by kmaehashi** [comment]
I've pushed docstring fixes!
/test

**Pull request merged**